### DISCUSSION TITLE
How to generate graphql-schema file from schema.prisma in Prisma3

### DISCUSSION THREAD

**Discussion by ahmed-trailfive** [prisma/prisma#13008] [Q&A] [answered]
**First Version of application using Prisma1 with GraphQL-Yoga**

In Prisma1 the following code was used to generate "prisma.graphql" file.
```
generate:
  - generator: graphql-schema
    output: ./generated/prisma.graphql
```
The file named "schema.graphql" is used to define the schema's with types imported from "prisma.graphql".

The graphQL-Yoga server was used and was initialized

```
const server = new GraphQLServer({
  typeDefs: typeDefs,
  resolvers)}
```
**While Upgrading code to Prisma3 from prisma1 **
The same generator is not available. 

So what should I do? 
Write the resolvers again in typegraphQL or Apollo?
Or the file "schema.graphql" can be used again?

![schema.graphql](https://user-images.githubusercontent.com/97940796/165287480-61368c6f-5715-453e-9f5c-f9ce5f16b011.png)

**Comment by austincrim** [comment, marked as answer]
Hey @ahmed-trailfive 👋 ,

As of Prisma 2, we are no longer directly responsible for generating and integrating with your GraphQL setup. I would encourage you to take a look at one of the examples [in our docs](https://www.prisma.io/docs/concepts/overview/prisma-in-your-stack/graphql#prisma--graphql-examples) that shows how to integrate with GraphQL from Prisma 2 and onward.

Let me know if you have specific questions as you get into it!

**Reply by ahmed-trailfive** [reply]
Thank-you @austincrim for replying. We are upgrading our system that was based on Prisma1 and GraphQL-Yoga v1. Stuck from 3 days now.😣.  I am gone through the docs and followed the [upgrade docs](https://www.prisma.io/docs/guides/upgrade-guides/upgrade-from-prisma-1/how-to-upgrade). Everything's worked perfect with Prisma up-gradation.

If you can just answer that if there is any alternative generator for Prisma1  "generator: graphql-schema"  in prisma2 onward? If not then do we have to rewrite the schema and resolvers written previously using graphQL-yoga V1?

**Reply by austincrim** [reply]
Prisma will not resolve relations for your implicitly now, you will need to write out your resolvers explicitly. You should be able to pull and reuse your GraphQL schema though, just with all types inlined instead of imported.

Take a look at Step 1 and 2 [here](https://www.prisma.io/docs/guides/upgrade-guides/upgrade-from-prisma-1/upgrading-prisma-binding-to-sdl-first#1-adjusting-your-graphql-schema) and let me know if that helps.

**Reply by ahmed-trailfive** [reply]
Okie. Thankyou. Will get back soon.

**Reply by ahmed-trailfive** [reply]
Thank you @austincrim it worked 🎉.
Just one more question. Should we keep our system in this structure or design graphQL using "typegraphQL" or "nexus"?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
